telegram/_message.py

```diff
@@ -826,11 +826,19 @@ def id(self) -> int:  # pylint: disable=invalid-name
     def link(self) -> Optional[str]:
         """:obj:`str`: Convenience property. If the chat of the message is not
         a private chat or normal group, returns a t.me link of the message.
+
+            .. versionchanged:: NEXT.VERSION
+                Added support for topics.
         """
         if self.chat.type not in [Chat.PRIVATE, Chat.GROUP]:
             # the else block gets rid of leading -100 for supergroups:
             to_link = self.chat.username if self.chat.username else f"c/{str(self.chat.id)[4:]}"
-            return f"https://t.me/{to_link}/{self.message_id}"
+            baselink = f"https://t.me/{to_link}/{self.message_id}"
+
+            # adds the thread for topics and replies
+            if (self.is_topic_message and self.message_thread_id) or self.reply_to_message:
+                baselink = f"{baselink}?thread={self.message_thread_id}"
```

Comment:
`self.reply_to_message` can be `None` without `self.message_thread_id` being present though, right? 🤔

Reply:
I'm too dumb to parse your rethorical question..
If it's a topic reply (with a `message_thread_id`), or it's a reply to something, we enhance the link with `thread`.
If it's not any of those things, we don't.

Is my logic wrong?

Reply:
My point is that a message can be a reply even if it's not in a forum group. To my understanding, `message_thread_id` is set only in froum groups. But maybe I'm mistaken …

Reply:
I think that `message_thread_id` is there even for replies to normal messages in a supergroup, pointing to the same `reply_to_message.id`.